src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -166,6 +157,47 @@ public BuildData(AbstractBuild<?, ?> build, Date currentTime) {
     }
   }
 
+  // Pipeline project build
+  public BuildData(Run<?, ?> build, Date currentTime, TaskListener listener) {
+    initData(build, currentTime);
+
+    Executor executor = build.getExecutor();
+    if (executor == null) {
+      buildHost = "master";
+    } else {
+      buildHost = StringUtils.isBlank(executor.getDisplayName()) ? "master" : executor.getDisplayName();
+    }
+
+    rootProjectName = projectName;
+    rootProjectDisplayName = displayName;
+    rootBuildNum = buildNum;
+
+    try {
+      buildVariables = build.getEnvironment(listener);
```

Comment:
How are sensitive variables to be handled in pipeline jobs?

`getSensitiveBuildVariables()` is only defined in `AbstractBuild`, so how is pipeline handling plugins like credentials binding?